### PR TITLE
Assigment 3 Kasey Miller

### PR DESCRIPTION
**Pull request by 1Kmiller1**
fixed 8.0net and got joke

### CODE REVIEW

**Comment by 1Kmiller1** [comment]
Assignment
For this assignment do the following:

Create two interfaces. One to represent the JokeService and one interface to represent outputting the joke to the screen (Console.WriteLine). ✔
The JokeService will need to have the interface applied to it. ✔
Create an implementation of the output interface that writes out the joke to the console. ✔
Implement the Jester class. It should take in both interfaces as dependencies. These dependencies should be null checked. ✔
The Jester class TellJoke() method should retrieve a joke from the JokeService. If the joke contains "Chuck Norris", skip it and get another. The joke should be written to the output dependency. ✔
Unit test the Jester class. Code coverage should be above 90% for this class. Moq will make this significantly easier. ✔

Fundamentals

Be sure you enable:
Nullability for all projects ✔
Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
Ensure that you turn on code analysis for all projects(EnableNETAnalyzers) ✔
Ensure that you turn on treat warnings as errors for all projects ✔
Ensure that you turn on CodeAnalysisTreatWarningsAsErrors ✔
Ensure that you turn on EnforceCodeStyleInBuild ✔
For this assignment, always use Assert.Equal<T>() (the generic version) ✔
Ensure there are no errors or warnings (including code analysis warnings) ✔
All of the above should be unit tested ✔
Choose simplicity over complexity ✔

Extra Credit

Unit test your implementation that writes the joke out to the screen. How hard could it be to unit test a single line method ;)✔

The Geek jokes API that is being used can also return jokes in a JSON format. Update the JokeService to retrieve jokes using JSON. GetJoke should still return a string. ❌

File: CanHazFunny/.editorconfig
Comment (AndrewKorchemniy) [inline CanHazFunny/.editorconfig:1]: We shouldn't be disabling warnings unless we're certain it provides no value.

File: CanHazFunny/CanHazFunny.Tests/JokeServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        string expectedJoke = "*Refrence to Other Funny Joke*";
+
+        mockjokeService.Setup(service => service.GetJoke()).Returns(expectedJoke);
+        //Act
+        string retreviedJoke = mockjokeService.Object.GetJoke();
+
+        //Assert
+        Assert.Equal(expectedJoke, retreviedJoke);
+    }
```
Comment (AndrewKorchemniy) [inline CanHazFunny/CanHazFunny.Tests/JokeServiceTest.cs:27]: This is testing the mocked objected and not the implementation.

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+{
+    private IJokeService _jokeService;
+    private IOutputService _outputService;
+
+    public Jester(IJokeService? jokeService, IOutputService? outputService) 
+    { 
+    
+        _jokeService = jokeService ?? throw new ArgumentNullException (nameof(jokeService));
+        _outputService = outputService ?? throw new ArgumentNullException (nameof (outputService));
+
```
Comment (AndrewKorchemniy) [inline CanHazFunny/CanHazFunny/Jester.cs:19]: We shouldn't allow users to inject nullable dependencies and then immediately throw if they do.
https://stevetalkscode.co.uk/null-injection

File: CanHazFunny/CanHazFunny/Program.cs [hunk 3]
```diff
@@ -1,10 +1,10 @@
 ﻿namespace CanHazFunny;
 
-class Program
+public static class Program
```
Comment (AndrewKorchemniy) [inline CanHazFunny/CanHazFunny/Program.cs:3]: Only main() itself needs to be static.

**Review by AndrewKorchemniy — COMMENTED** [review]
late assignment

**Comment by Joshua-Lester3** [comment]
Hey Kasey, just a heads up: you'll need to find a partner as that's a requirement for the assignments.

**Pull request closed without merging**